infra/RepoManager.py

```diff
@@ -0,0 +1,209 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Class to manage a git repository via python.
+
+This class is to be used to implement git commands over
+a python API and manage the current state of the git repo.
+
+  Typical usage example:
+
+    r_man =  RepoManager('https://github.com/google/oss-fuzz.git')
+    r_man.checkout('5668cc422c2c92d38a370545d3591039fb5bb8d4')
+    r_man.close()
+"""
+import os
+import shutil
+import subprocess
+
+
+class RepoManagerError(Exception):
+  """Class to describe the exceptions in RepoManager."""
+
+  def __init__(self, message):
+    """ Init the exception class.
+
+    Args:
+      message: the message to propagate to user
+    """
+    super().__init__(message)
+
+
+class RepoManager(object):
+  """Class to manage git repos from python.
+
+  Attributes:
+    repo_url: The location of the git repo
+    base_dir: The location of where the repo clone is stored locally
+    repo_name: The name of the github project
+    repo_dir: The location of the main repo
+    full_path: The full filepath location of the main repo
+  """
+
+  def __init__(self, repo_url, commit=None, base_dir='tmp'):
+    """Constructs a repo manager class.
+
+    Args:
+      repo_url: The github url needed to clone
+      commit: The specified commit to be checked out
+      base_dir: The local location the repo will live in
+    """
+
+    self.repo_url = repo_url
+    self.base_dir = base_dir
+    self.repo_name = self.repo_url.split('/')[-1].strip('.git')
+    self.repo_dir = os.path.join(self.base_dir, self.repo_name)
+    self.full_path = os.path.join(os.getcwd(), self.repo_dir)
+    self._clone()
+
+    if commit:
+      self.checkout_commit(commit)
+
+  def _clone(self):
+    """Creates a clone of the repo in the specified directory.
+
+      Raises:
+        RepoManagerError if the repo was not able to be cloned
+    """
+    if not os.path.exists(self.base_dir):
+      os.makedirs(self.base_dir)
+    self.remove_repo()
+    _, err = self._run_command(['git', 'clone', self.repo_url], self.base_dir)
+    if err is not None:
+      raise RepoManagerError(
+          'Failed cloning repo %s, with error %s)' % (self.repo_url, err))
+    if not self._is_git_repo():
+      raise RepoManagerError('%s is not a git repo' % self.repo_url)
+
+  def _run_command(self, command, location='.'):
+    """ Runs a shell command in the specified directory location.
+
+    Args:
+      command: The command as a list to be run
+
+    Returns:
+      The stdout of the command, the stderr of the command
+    """
+    process = subprocess.Popen(command, stdout=subprocess.PIPE, cwd=location)
+    out, err = process.communicate()
+    if err is not None:
+      err = err.decode('ascii')
+      print('Error %s running command %s' % (err, command))
```

Comment:
Let's use the logging module instead of prints. This is especially useful on infra where e.g. on GCP this will be automatically reported as an error.